blockprod/src/detail/mod.rs

```diff
@@ -305,17 +307,14 @@ impl BlockProduction {
             Arc::new(AcqRelAtomicU64::new(tip_plus_one.as_int_seconds()))
         };
 
-        let max_block_timestamp = {
-            let current_timestamp =
-                BlockTimestamp::from_duration_since_epoch(self.time_getter().get_time());
-
-            current_timestamp
-                .add_int_seconds(self.chain_config.max_future_block_time_offset().as_secs())
-                .ok_or(ConsensusCreationError::TimestampOverflow(
-                    current_timestamp,
-                    self.chain_config.max_future_block_time_offset().as_secs(),
-                ))?
-        };
+        let min_block_timestamp =
+            BlockTimestamp::from_duration_since_epoch(self.time_getter().get_time());
```

Comment:
This is not really correct. The minimum timestamp value is the previous block's timestamp + 1.

Reply:
Unless I'm misunderstanding the meaning of this...

Reply:
I've just gone through the whole change now. I see... I think the name `min_block_timestamp` is wrong, and should be named something like `min_verification_block_timestamp` so it doesn't get confused with the "minimum possible block timestamp"

Reply:
Yea it's a bit subtle. Basically it's the timestamp where the search for a block starts. Set it too low and node operator will miss out on fees because mempool will not include some locked transactions. Set it to high and it will not be correct. An alternative is to construct the transaction sequence from the scratch every time block production attempts to construct a block with a different timestamp. Will update the name and put a comment explaining it.

Reply:
Well, the timestamps are decided before any transactions are added to a block. So there's no way really to optimize this, assuming that the block reward is going to be higher than any transaction fee. Please ignore the game-theoretic aspect of optimizing locked outputs for fees for timestamps because the timestamps' time slots are mostly empty during normal operation with maximum probability of 1/120 per total, assuming the user has 100% of the stake.

Reply:
Added some explanation comments to this bit.